### PR TITLE
20200623 site term

### PR DESCRIPTION
**Pull request by mmpagani**
This PR adds a modifiable GMM that incorporates the site term of CY2014. This modifiable GMPE can be used to add a site term to GMMs that do not include it natively. It's not the best option but just a solution in absence of more advanced models. In principle, it works with all the GMPEs in the OQ Engine suite, provided that their reference rock conditions are close to 1100 m/s. The original GMPEs must have a DEFINED_FOR_REFERENCE_VELOCITY attribute.

### CODE REVIEW

**Comment by micheles** [comment]
I guess `gmpe_name` can be different from ChiouYoungs2014, otherwise there would be no point in passing it. Perhaps in the docstring you should write down which are the GMPEs modifiable with this site term, to give some guidance to the users.

File: openquake/hazardlib/gsim/mgmpe/cy14_site_term.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Module :mod:`openquake.hazardlib.mgmp.nrcan15_site_term` implements
+:class:`~openquake.hazardlib.mgmpe.ask14_site_term.ASK14SiteTerm`
```
Comment (rcgee) [inline openquake/hazardlib/gsim/mgmpe/cy14_site_term.py:19]: let's update this part?

File: openquake/hazardlib/gsim/mgmpe/cy14_site_term.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        for spec of input and result values.
+        """
+
+        # Prepare sites
+        sites_rock = copy.copy(sites)
+        sites_rock.vs30 = np.ones_like(sites_rock.vs30) * 1130.
+
+        # Compute mean and standard deviation using the original GMM. These
+        # values are used as ground-motion values on reference rock conditions.
+        # CHECK [MP]: The computed reference motion is equal to the one in the
```
Comment (rcgee) [inline openquake/hazardlib/gsim/mgmpe/cy14_site_term.py:85]: planning to check? or this was checked?

File: openquake/hazardlib/tests/gsim/mgmpe/cy14_site_term_test.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        np.testing.assert_almost_equal(mean, mean_expected, decimal=7)
+
+        # Here we use a quite large tolerance since in the site term we take
+        # the std from the calculation of motion on reference rock. This
+        # does not match the std that the same GMM computes for soft soils
+        # with the same remaining conditions
+        np.testing.assert_almost_equal(stds, stds_expected, decimal=1)
+
+    def test_raise_error(self):
+        """ raise error when DEFINED_FOR_REFERENCE_VELOCITY misses """
```
Comment (rcgee) [inline openquake/hazardlib/tests/gsim/mgmpe/cy14_site_term_test.py:209]: what is meant by "misses"?

File: openquake/hazardlib/gsim/mgmpe/cy14_site_term.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+import numpy as np
+from openquake.hazardlib import const
+from openquake.hazardlib.gsim.base import GMPE, registry
+from openquake.hazardlib.gsim.chiou_youngs_2014 import ChiouYoungs2014
+
+
+class CY14SiteTerm(GMPE):
+    """
+    Implements a modified GMPE class that can be used to account for local
+    soil conditions in the estimation of ground motion.
```
Comment (rcgee) [inline openquake/hazardlib/gsim/mgmpe/cy14_site_term.py:32]: Could add to the end of this sentence: " using the site term from ChiouYoungs2014"

**Comment by rcgee** [comment]
Apart from minor comments, LGTM. Thanks Marco!

**Pull request merged**